ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -122,10 +125,11 @@ void *libX11_Variadic_u64(uint64_t a_0, size_t count, unsigned long *list, void
       # x1 = <count>
       # x9 = <list ptr>
 
-      # Stack objects
-      # Count >= 7
+      # The number of arguments on the stack will always be (Count - 7) + 1
+
       # Subtract 6 count objects
-      # Leaves us at least 1 (nullptr)
+      # Gives us the total number of objects that need to be stored on to the stack.
+      # If exactly 7 objects then only nullptr ends up on the stack.
       sub x1, x1, 6
 
       # Round up to the nearest pair
```

Comment:
Why do we need to do this? Is it an expectation by X11 libraries, or something about the specifics in the AArch64 ABI?

Reply:
AArch64 ABI requires rsp to be 16-byte aligned.